### PR TITLE
KAFKA-16904: Metric to measure the latency of remote read requests

### PR DESCRIPTION
**Pull request by kamalcph**
This metric will be useful to measure the time taken to fetch data from remote storage.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by satishd** [comment]
@kamalcph Please resolve the merge conflicts.

**Comment by kamalcph** [comment]
@satishd @showuon 

Rebased the changes against trunk. PTAL.

**Review by satishd — COMMENTED** [review]
Thanks @kamalcph for the PR. Overall LGTM.

How is KAFKA-15776 related to`RemoteLogReaderFetchRateAndTimeMs` metric?

**Comment by kamalcph** [comment]
> Thanks @kamalcph for the PR. Overall LGTM.
> 
> How is [KAFKA-15776](https://issues.apache.org/jira/browse/KAFKA-15776) related to`RemoteLogReaderFetchRateAndTimeMs` metric?

We added it to the [KIP-1018](https://cwiki.apache.org/confluence/display/KAFKA/KIP-1018%3A+Introduce+max+remote+fetch+timeout+config+for+DelayedRemoteFetch+requests) as part of the [review](https://lists.apache.org/thread/9x21hzpxzmrt7xo4vozl17d70fkg3chk).

**Review by satishd — COMMENTED** [review]
Thanks @kamalcph for the reply. I prefer having a separate JIRA instead of pointing to the root JIRA for the related feature.

Please file a separate JIRA to update the documentation with the new metric. 

It is good to add a test that ensures the added metric is getting removed whenever RLM is closed.

**Review by showuon — COMMENTED** [review]
LGTM! Could we add a test to verify the metric is updated as expected? Maybe you can add some verification in `ReplicaManagerTest#testRemoteLogReaderMetrics` directly?

**Comment by kamalcph** [comment]
@satishd @showuon 

Addressed your review comments. PTAL.

File: core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala [hunk 1]
```diff
@@ -4167,6 +4170,8 @@ class ReplicaManagerTest {
       assertEquals(3, yammerMetricValue("RemoteLogReaderTaskQueueSize").asInstanceOf[Int])
       // unlock all tasks
       doneLatch.countDown()
+      responseLatch.await()
```
Comment (showuon) [inline core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala]: nit: Please add a time limit to this await.

**Review by showuon — APPROVED** [review]
LGTM!

**Review by showuon — APPROVED** [review]
LGTM! Thanks.

**Review by clolov — APPROVED** [review]
Thank you for the metric!

**Comment by showuon** [comment]
Re-triggering CI build: https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-16209/8/

**Review by satishd — APPROVED** [review]
Thanks @kamalcph for addressing the review comments. LGTM.

**Comment by kamalcph** [comment]
Failed tests are unrelated.

**Pull request merged**